include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1434,101 +1442,110 @@ namespace internal
 
       // ----------------------- prolongation matrix (1) -----------------------
       {
-        AssertDimension(fe_fine.n_base_elements(), 1);
-        if (reference_cell == ReferenceCells::get_hypercube<dim>())
+        if (fe_fine.n_base_elements() == 1 &&
+            fe_fine.base_element(0).n_components() == 1)
           {
-            const auto fe = create_1D_fe(fe_fine.base_element(0));
+            if (reference_cell == ReferenceCells::get_hypercube<dim>())
+              {
+                const auto fe = create_1D_fe(fe_fine.base_element(0));
 
-            std::vector<unsigned int> renumbering(fe->n_dofs_per_cell());
-            {
-              AssertIndexRange(fe->n_dofs_per_vertex(), 2);
-              renumbering[0] = 0;
-              for (unsigned int i = 0; i < fe->dofs_per_line; ++i)
-                renumbering[i + fe->n_dofs_per_vertex()] =
-                  GeometryInfo<1>::vertices_per_cell * fe->n_dofs_per_vertex() +
-                  i;
-              if (fe->n_dofs_per_vertex() > 0)
-                renumbering[fe->n_dofs_per_cell() - fe->n_dofs_per_vertex()] =
-                  fe->n_dofs_per_vertex();
-            }
+                std::vector<unsigned int> renumbering(fe->n_dofs_per_cell());
+                {
+                  AssertIndexRange(fe->n_dofs_per_vertex(), 2);
+                  renumbering[0] = 0;
+                  for (unsigned int i = 0; i < fe->dofs_per_line; ++i)
+                    renumbering[i + fe->n_dofs_per_vertex()] =
+                      GeometryInfo<1>::vertices_per_cell *
+                        fe->n_dofs_per_vertex() +
+                      i;
+                  if (fe->n_dofs_per_vertex() > 0)
+                    renumbering[fe->n_dofs_per_cell() -
+                                fe->n_dofs_per_vertex()] =
+                      fe->n_dofs_per_vertex();
+                }
```

Comment:
I know this is pre-existing, but isn't this the regular hierarchic-to-lexicographic renumbering you would get from `ShapeInfo::lexicographic_numbering` or maybe the inverse? I know it is a bit of a hammer to call that function, but this is not performance-critical code and 1D anyway, and I find it more documenting, and is what we do below.